File: CodingSamples/DotNet/Advanced/Networking/TcpClientTest/DemoApp2/Models/ItemInfo.cs
namespace DemoApp.Models;

public readonly record struct ItemInfo(double Cost, int Stock)
{
    public static ItemInfo Parse(string text)
    {
        string[] segments = text.Split('&');
        double p = double.Parse(segments[0][5..]);
        int n = int.Parse(segments[1][6..]);
        return new ItemInfo(p, n);
    }
}

File: CodingSamples/Java/Advanced/Networking/TCPClientTest/demo1/src/main/java/app/models/ItemInfo.java
package app.models;

public record ItemInfo(double cost, int stock) {
    
    public static ItemInfo parse(String text) {
        String[] segments = text.split("&");
        double p = Double.parseDouble(segments[0].substring(5));
        int n = Integer.parseInt(segments[1].substring(6));
        return new ItemInfo(p, n);
    }
}
